### PR TITLE
add changelog for 1.4.0

### PR DESCRIPTION
**Pull request by thehowl**
mostly generated straight out of the tool, but removed all refactoring and testing PRs. also added link to the blog on top.

### CODE REVIEW

**Comment by strk** [comment]
ouch it's huge, I liked the blog post version much more, too much information becomes noise (which tool, btw?)

**Comment by lafriks** [comment]
Docs should probably be removed

**Comment by thehowl** [comment]
@strk https://github.com/lunny/changelog

As for the fact that it's huge - I agree. I'm now trying to further cut some some more entries. I quite agree with you in saying that the other changelogs should be more descriptive and shorter, but I also see the point in keeping it "technical" and without hiding too much of what has been changed (for instance, one might not find that we've fixed that very specific bug in the blog post, but they can do so in the CHANGELOG file).

@lafriks I agree, they're not that important. Done.

**Review by lafriks — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,112 @@
 # Changelog
 
+This changelog goes through all the changes that have been made in each release
+without substantial changes to our git log; to see the highlights of what has
+been added to each release, please refer to the [blog](https://blog.gitea.io).
+
+## [1.4.0-rc1](https://github.com/go-gitea/gitea/releases/tag/v1.4.0-rc1) - 2018-01-31
```
Comment (appleboy) [inline CHANGELOG.md:7]: Add new line here?

Comment (lafriks) [reply CHANGELOG.md:7]: previously we did not do that

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,112 @@
...
+  * Reduce overhead of upgrades for users with custom stylesheets/JS (#3051)
+  * Default log level to Info without hardcoding it in installer (#3041)
+  * Memory usage improvements (#3013)
+  * Add fingerprint to ssh key endpoints. (#3009)
+  * Improve memory usage when reaching diff limits (#2990)
+  * Expandable commit bodies (#2980)
+  * Update gitgraph.js to fix blurry commit graph on HiDPI screens (#2957)
+  * Fix language names (#2955)
+  * Remove render issue link (#2954)
+  * Git LFS lock api (#2938)
```
Comment (sapk) [inline CHANGELOG.md]: I think it should be in features as it permit to use the command `git lfs lock`.

**Review by sapk — APPROVED** [review]

**Comment by lafriks** [comment]
~~Waiting for Swedish translation to be approved~~

**Pull request merged**